backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -44,12 +49,18 @@ public boolean isDedicatedCpuPinningPossibleAtHost(Map<Guid, List<VdsCpuUnit>> v
         }
 
         previewPinOfPendingExclusiveCpus(cpuTopology, vmToPendingPinnings, vm.getCpuPinningPolicy());
+        filterBadNumaMemorySockets(cpuTopology, vm, hostId);
 
         int socketsLeft = vm.getNumOfSockets();
+        int highestAmountOfNumaNodesInSocket = Integer.MAX_VALUE;
 
         for (int socket : getOnlineSockets(cpuTopology)) {
+            if (!vm.getvNumaNodeList().isEmpty()) {
+                highestAmountOfNumaNodesInSocket = getVirtualNumaNodesInSocket(cpuTopology, hostId, socket);
+            }
```

Comment:
I think this leaves room for issues because highestAmountOfNumaNodesInSocket is used for all sockets.. why not to say as you did below - if there are cores then check if there are vnuma nodes and if so do the Math.min thingie?

Reply:
here it simplier, on each socket we refresh the `highestAmountOfNumaNodesInSocket` per the current socket we are with.
if there is no vNUMA it will be `Integer.MAX_VALUE` and won't be changed. i don't see the problem here